security/pfSense-pkg-snort/files/usr/local/pkg/snort/snort.inc

```diff
@@ -125,12 +118,10 @@ function snort_get_blocked_ips() {
 }
 
 function snort_generate_id() {
-	global $config;
 
-	$snortglob = $config['installedpackages']['snortglobal']['rule'];
 	while (true) {
 		$snort_uuid = mt_rand(1, 65535);
-		foreach ($snortglob as $value) {
+		foreach (config_get_path('installedpackages/snortglobal/rule') as $value) {
```

Comment:
You should add a default value of an empty array here, that way if the configuration doesn't exist, the `foreach` won't fail.

For example `config_get_path('installedpackages/snortglobal/rule', [])`

Reply:
Added default return value.